src/main/java/cubicchunks/world/OpacityIndex.java

```diff
@@ -34,210 +35,287 @@
 
 public class OpacityIndex implements IOpacityIndex {
 
-	private static int None = Integer.MIN_VALUE;
+	/**
+	 * Special value to indicate the absence of a segment in the minimum and maximum arrays.
+	 */
+	private static final int NONE = Integer.MIN_VALUE;
 
-	// it's hard to find a segment value we can't use
-	// this is the best I can do
-	// a 1-block segment at the very top of the world with opacity zero
-	// seems a bit redundant =P
-	private static int NoneSegment = packSegment(0x7fffff, 0);
+	/**
+	 * Special value to indicate the absence of a segment in the segments arrays.
+	 * Since all integer values might represent a valid segment, an unlikely value to occur has been chosen. It
+	 * logically represents a 1-block segment at the very top of the world with opacity zero.
+	 */
+	private static final int NONE_SEGMENT = packSegment(0x7fffff, 0);
 
-	private int[] m_ymin;
-	private int[] m_ymax;
 	/**
-	 * The array of segments for each x/z position stores Y positions at which the opaque/transparent state changes.
+	 * Array containing the y-coordinates of the lowest segment in each block column. The value {@link #NONE} is used if
+	 * a given block column does not contain any segments.
 	 */
-	private int[][] m_segments;
+	private final int[] ymin;
 
-	private int heightMapLowest = None;
+	/**
+	 * Array containing the y-coordinate of the highest segment in each block column. The value {@link #NONE} is used if
+	 * a given block column does not contain any segments.
+	 */
+	private final int[] ymax;
+
+	/**
+	 * Array containing an array of segments for each x/z position in a column.
+	 */
+	private final int[][] segments;
+
+	private int heightMapLowest;
+
+	private int hash;
+
+	private boolean needsHash;
 
-	private int m_hash;
-	private boolean m_needsHash;
 
 	public OpacityIndex() {
-		m_ymin = new int[16*16];
-		m_ymax = new int[16*16];
-		m_segments = new int[16*16][];
+		this.ymin = new int[Cube.SIZE * Cube.SIZE];
+		this.ymax = new int[Cube.SIZE * Cube.SIZE];
+
+		this.segments = new int[Cube.SIZE * Cube.SIZE][];
 
 		// init to empty
-		for (int i = 0; i < 16*16; i++) {
-			m_ymin[i] = None;
-			m_ymax[i] = None;
+		for (int i = 0; i < Cube.SIZE * Cube.SIZE; i++) {
+			this.ymin[i] = NONE;
+			this.ymax[i] = NONE;
 		}
 
-		m_hash = 0;
-		m_needsHash = true;
+		this.heightMapLowest = NONE;
+		this.hash = 0;
+		this.needsHash = true;
 	}
 
+
+	private static int packSegment(int pos, int opacity) {
+		return Bits.packUnsignedToInt(opacity, 8, 24) | Bits.packSignedToInt(pos, 24, 0);
+	}
+
+	private static int unpackOpacity(int packed) {
+		return Bits.unpackUnsigned(packed, 8, 24);
+	}
+
+	private static int unpackPosition(int packed) {
+		return Bits.unpackSigned(packed, 24, 0);
+	}
+
+	private static int getLastSegmentIndex(int[] segments) {
+		for (int i = segments.length - 1; i >= 0; i--) {
+			if (segments[i] != NONE_SEGMENT) {
+				return i;
+			}
+		}
+		throw new Error("Invalid segments state");
+	}
```

Comment:
Why add all of these methods at the top? They just seem to get in the way of seeing what the class actually does. Their names are obvious enough that you don't need to see the code in them.